pkg/lib/cockpit-components-terminal.jsx

```diff
@@ -261,16 +340,17 @@ export class Terminal extends React.Component {
     }
 
     componentWillUnmount() {
-        this.disconnectChannel();
-        this.terminal.dispose();
         window.removeEventListener('resize', this.onWindowResize);
         this.onFocusOut();
+        this.unmountTerminal(this.terminal_state);
+        if (!this.props.state)
+            this.terminal_state.close();
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-21680-6bb6065b-20250807-121948-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)